examples/ping-pong.hs

```diff
@@ -38,50 +43,158 @@ startHandler :: DiscordHandler ()
+                    )
+                )
+                Nothing
+                Nothing
+                Nothing
+                Nothing
+            )
+        )
+    )
 
 -- If an event handler throws an exception, discord-haskell will continue to run
 eventHandler :: Event -> DiscordHandler ()
 eventHandler event = case event of
-      MessageCreate m -> when (not (fromBot m) && isPing m) $ do
-        void $ restCall (R.CreateReaction (messageChannel m, messageId m) "eyes")
-        threadDelay (2 * 10^(6 :: Int))
-
-        -- A very simple message.
-        void $ restCall (R.CreateMessage (messageChannel m) "Pong!")
-
-        -- A more complex message. Text-to-speech, does not mention everyone nor
-        -- the user, and uses Discord native replies.
-        -- Use ":info" in ghci to explore the type
-        let opts :: R.MessageDetailedOpts
-            opts = def { R.messageDetailedContent = "Here's a more complex message, but doesn't ping @everyone!"
-                       , R.messageDetailedTTS = True
-                       , R.messageDetailedAllowedMentions = Just $
-                          def { R.mentionEveryone = False
-                              , R.mentionRepliedUser = False
-                              }
-                       , R.messageDetailedReference = Just $
-                          def { referenceMessageId = Just $ messageId m }
-                       }
-        void $ restCall (R.CreateMessageDetailed (messageChannel m) opts)
-      _ -> return ()
+  MessageCreate m -> when (not (fromBot m) && isPing m) $ do
+    void $ restCall (R.CreateReaction (messageChannel m, messageId m) "eyes")
+    threadDelay (2 * 10 ^ (6 :: Int))
+
+    -- A very simple message.
+    void $ restCall (R.CreateMessage (messageChannel m) "Pong!")
+
+    -- A more complex message. Text-to-speech, does not mention everyone nor
+    -- the user, and uses Discord native replies.
+    -- Use ":info" in ghci to explore the type
+    let opts :: R.MessageDetailedOpts
+        opts =
+          def
+            { R.messageDetailedContent = "Here's a more complex message, but doesn't ping @everyone!",
+              R.messageDetailedTTS = True,
+              R.messageDetailedAllowedMentions =
+                Just $
+                  def
+                    { R.mentionEveryone = False,
+                      R.mentionRepliedUser = False
+                    },
+              R.messageDetailedReference =
+                Just $
+                  def {referenceMessageId = Just $ messageId m}
+            }
+    void $ restCall (R.CreateMessageDetailed (messageChannel m) opts)
+  Ready _ _ _ _ _ _ pa@(PartialApplication i _) ->
+    trace
+      (show pa)
+      ( restCall
+          ( R.CreateGuildApplicationCommand i serverid exampleSlashCommand
+          )
+      )
+      >>= \rs -> trace (show rs) (return ())
+  InteractionCreate i@Interaction {..} -> trace (show i) $ do
+    let cid = fromJust interactionChannelId
```

Comment:
Is `fromJust` safe here? A crash in an event handler is ignored so in general yes, but what's the particular thought here?

Reply:
The thought was to give a quick and dirty demonstration of how interactions could work, although I accept that this is unsafe (as is `head`).

I can come up with safer alternatives, but I didn't want to have to mess with the monads.

Reply:
Yeah it's not necessary, but I prefer to have good practices in examples. Don't have to change it now